apps/assets/tasks/ping.py

```diff
@@ -48,3 +53,38 @@ def test_node_assets_connectivity_manual(node):
     asset_ids = [str(i) for i in asset_ids]
     org_id = str(current_org.id)
     return test_assets_connectivity_task.delay(asset_ids, org_id, task_name)
+
+
+@shared_task(
+    verbose_name=_('Test assets port connectivity'),
+    activity_callback=lambda self, asset_ids, port, *args, **kwargs: (asset_ids,),
+    description=_(
+        "When clicking 'Test Asset Port Connectivity' in 'Asset List' this task will be executed"
+    )
+)
+def test_assets_port_connectivity(asset_ids, port):
+    log_path = get_celery_task_log_path(test_assets_port_connectivity.request.id)
+    printer = ColoredFilePrinter(open(log_path, 'a'), flush_now=True)
+    assets = Asset.objects.filter(id__in=asset_ids).values('name', 'address')
+    msg = f"{_('Asset')}: %s {_('Test port')} {port}: %s"
+    success, failed = [], []
+    for asset in assets:
+        asset_display = f'{asset["name"]}({asset["address"]})'
+        try:
+            socket.create_connection(
+                (asset["address"], port), timeout=10, all_errors=True
+            )
+            printer.green(msg % (asset_display, _('Success')))
+            success.append(asset)
+        except ExceptionGroup:
+            printer.red(msg % (asset_display, _('Failed')))
+            failed.append(asset)
+    printer.line()
+    printer.info(
+        _('Total {}, success {}, failure {}').format(len(assets), len(success), len(failed))
+    )
+
+
+def test_assets_port_connectivity_manual(assets, port):
+    asset_ids = [str(i.id) for i in assets]
+    return test_assets_port_connectivity.delay(asset_ids, port)
```

Comment:
The provided code appears to be an automation task script written for testing the connection of certain assets. 

Here are the main differences noted:

- The `# ~*~ coding: utf-8 ~*~` line is not necessary and can be removed since Python 3 supports UTF-8 by default with no need for specific encoding.

- Import statements have been altered, but their functionality has not changed so it doesn't seem relevant here.

- There seems to be a typo in a variable name (`task_name`) which should be corrected according to the original source code.

No additional issues were recognized due to the lack of any actual discrepancies that typically require manual review for changes.

Potential improvements could include updating logging configurations if needed or changing comment structures based on project standards (e.g., adding more meaningful comments).

No further optimizations are suggested given there isn’t explicit commentary indicating where these scripts might need enhancement.